src/coreclr/minipal/minipal.h

```diff
@@ -76,3 +76,16 @@ class VMToOSInterface
     //  true if it succeeded, false if it failed
     static bool ReleaseRWMapping(void* pStart, size_t size);
 };
+
+#if defined(HOST_64BIT) && defined(FEATURE_CACHED_INTERFACE_DISPATCH)
```

Comment:
I see we're adding a HOST_64BIT guard here but when I look at the CMakeLists changes it doesn't seem like we're necessarily doing the work to enable 128-bit cmpxchg for all 64-bit targets. Is this going to be fine on all the 64-bit targets we support or do we need to guard it with more specific conditionals? I know we have some support for more esoteric architectures than arm64 and amd64, and I'm not sure whether those are 64-bit hosts.

64-bit WASM doesn't have 16-byte cmpxchg, but it's probably safe to ignore that one since most WASM applications are using a 32-bit address space (it's much faster) where an 8-byte cmpxchg is good enough

Reply:
It was discussed above https://github.com/dotnet/runtime/pull/111771#discussion_r1938186351, currently it's only enabled for amd64 and arm64 for non-nativeaot parts. Also, amd64 has `-mcx16` flag added as part of this PR; which exists in nativeaot where this code is copied from.